src/rgw/support/rgw_support_bundle

```diff
@@ -166,8 +162,40 @@ class RGWSupportBundle:
         if coredumps:
             crash_dump_dir = os.path.join(log_dir, 'rgw_debug')
             if os.path.exists(crash_dump_dir):
-                shutil.copytree(crash_dump_dir,\
-                    os.path.join(RGWSupportBundle._tmp_src, 'rgw_debug'))
+                if size_limit:
+                    latest_5_cores = RGWSupportBundle._get_latest_files(
+                        directory=crash_dump_dir, number_of_files=5)
+                    for file_name in latest_5_cores:
+                        shutil.copyfile(os.path.join(crash_dump_dir, file_name),
+                            os.path.join(RGWSupportBundle._tmp_src, 'rgw_debug'))
+                else:
+                    shutil.copytree(crash_dump_dir,\
+                        os.path.join(RGWSupportBundle._tmp_src, 'rgw_debug'))
+
+        # remaining size limit in folder
+        remaining_size_limit = RGWSupportBundle._get_remaining_folder_size(
+            RGWSupportBundle._tmp_src, size_limit)
+        # apply truncate logic on all textual logs
+        if size_limit and not coredumps:
```

Comment:
if size limit is passed, a `tmp_dir` directory is created and files are truncated and kept in this directory, once all the files are truncated, original files are replaced with newly created files and the directory `tmp_dir` is removed